### PR TITLE
Do not verify on state import for relay chain verifier

### PR DESCRIPTION
**Pull request by skunert**
Follows: https://github.com/paritytech/substrate/pull/13058

After https://github.com/paritytech/cumulus/pull/1909 lands, we have warp sync for parachains.

The same fix as for aura needs to apply to the verifier that is used [here](https://github.com/paritytech/cumulus/blob/master/polkadot-parachain/src/service.rs#L1031) for parachains that do not have aura from the start. Otherwise, we would try to verify the warped-to block by using its parent (that we don't have).

### CODE REVIEW

**Comment by paritytech-cicd-pr** [comment]
The CI pipeline was cancelled due to failure one of the required jobs.
**Job name**: test-linux-stable
**Logs**: https://gitlab.parity.io/parity/mirrors/cumulus/-/jobs/2313569

**Review by bkchr — APPROVED** [review]

**Comment by bkchr** [comment]
BTW: https://github.com/paritytech/substrate/pull/13221

**Comment by skunert** [comment]
> BTW: [paritytech/substrate#13221](https://github.com/paritytech/substrate/pull/13221)

Just for clarity, as mentioned in the linked PR we are setting the fork choice rule in `ParachainBlockImport` [here](https://github.com/paritytech/cumulus/blob/07f5b79f56316b774a05db4a311e3e9ed40303f4/client/consensus/common/src/lib.rs#L151-L154). So this PR is still fine, correct?

File: client/consensus/relay-chain/src/import_queue.rs [hunk 1]
```diff
@@ -58,6 +58,18 @@ where
 		&mut self,
 		mut block_params: BlockImportParams<Block, ()>,
 	) -> Result<(BlockImportParams<Block, ()>, Option<Vec<(CacheKeyId, Vec<u8>)>>), String> {
+		if block_params.with_state() {
```
Comment (davxy) [inline client/consensus/relay-chain/src/import_queue.rs:61]: Super nit (feel free to ignore). In [this](https://github.com/paritytech/substrate/pull/13221) new PR we've grouped the checks together. If you like we can do the same here .

**Review by davxy — APPROVED** [review]
LGTM.

Kinda related... can I please have your feedback [here](https://github.com/paritytech/substrate/pull/13221#discussion_r1087635967). Ty

**Comment by skunert** [comment]
bot merge

**Pull request merged**